sbt-app/src/sbt-test/dependency-management/sbt-plugin-publish/test

```diff
@@ -0,0 +1,6 @@
+> example / checkPublishedArtifacts
+> example / checkPublish
+
+> testProject / checkUpdate
+> set testProject / useCoursier := false
+> testProject / checkUpdate
```

Comment:
Do we cover publishing to ivy repositories? (e.g. `publishLocal`)

Do we cover resolving a plugin that we just published?

Reply:
That's a great question; I thought the Ivy pattern and approach is entirely different from that of SBT+Maven?

Reply:
I expanded the scripted sbt-plugin-publish to test the publication to  `.ivy/local` and resolution from it.

For the Ivy publication, there are two cases, depending on the value of `publishMavenStyle`.
- `publishMavenStyle := false`: Strictly no changes, we publish:
```
org.example/example/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/
    ├── docs
    │   └── example-javadoc.jar
    ├── ivys
    │   └── ivy.xml
    ├── jars
    │   └── example.jar
    └── src
        └── example-sources.jar
```
- `publishMavenStyle.value := true`: We publish the legacy and new poms, and all the associated artifacts.
```
org.example/example/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/
    ├── docs
    │   ├── example_2.12_1.0-javadoc.jar
    │   └── example-javadoc.jar
    ├── ivys
    │   └── ivy.xml
    ├── jars
    │   ├── example_2.12_1.0.jar
    │   └── example.jar
    ├── poms
    │   ├── example_2.12_1.0.pom
    │   └── example.pom
    └── src
        ├── example_2.12_1.0-sources.jar
        └── example-sources.jar
```

In the test, we do a `publishLocal` on a project that is configured for Maven, so it is a Maven-style publication to Ivy.